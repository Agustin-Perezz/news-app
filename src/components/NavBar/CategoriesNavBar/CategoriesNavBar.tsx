import { useContext } from 'react';
import { Box, Tabs, ThemeProvider } from '@mui/material';
import { listCategories } from './listCategories';
import { CustomTab, tabFont } from './CustomStylesTab';
import { HandleClickProps } from '../../../types';
import { CategoryContext } from '../../../context/CategoryContext';

export const CategoriesNavBar: React.FC<HandleClickProps> = ({ handleClick }) => {
  const { activeCategory, handleSetTab } = useContext(CategoryContext);
  return (
    <Box sx={{ width: '100%' }}>
      <Box
        sx={{
          borderBottom: 1,
          borderColor: 'GrayText',
          justifyContent: 'center',
          margin: '0 auto',
          display: 'flex',
        }}
      >
        <ThemeProvider theme={tabFont}>
          <Tabs
            value={activeCategory}
            onChange={handleSetTab}
            aria-label="category-tabs"
            indicatorColor="secondary"
          >
            {listCategories.map(({ category, endpoint, url }) => (
              <CustomTab
                label={category}
                key={category}
                onClick={() => {
                  handleClick({
                    urlParameter: category,
                    endpoint,
                    navigateUrl: url,
                  });
                }}
              />
            ))}
          </Tabs>
        </ThemeProvider>
      </Box>
    </Box>
  );
};
