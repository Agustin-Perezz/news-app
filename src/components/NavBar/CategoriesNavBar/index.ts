export * from './CategoriesNavBar';
export * from './listCategories';
