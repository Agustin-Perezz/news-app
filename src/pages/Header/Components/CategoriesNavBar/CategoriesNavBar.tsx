import { Box, Tabs, ThemeProvider, Tab } from '@mui/material';
import { listCategories } from './listCategories';
import { CustomTab, tabFont } from './CustomStyleTab';
import { HandleChangeTabProps, HandleClickProps } from '../../../../types';

type Props = HandleClickProps & HandleChangeTabProps & { activeCategoryTab: number };

export const CategoriesNavBar: React.FC<Props> = ({
  handleClick,
  handleChangeTab,
  activeCategoryTab,
}) => {
  return (
    <Box sx={{ width: '100%' }}>
      <Box
        sx={{
          borderBottom: 1,
          borderColor: 'GrayText',
          justifyContent: 'center',
          margin: '0 auto',
          display: 'flex',
        }}
      >
        <ThemeProvider theme={tabFont}>
          <Tabs
            value={activeCategoryTab}
            onChange={handleChangeTab}
            aria-label="category-tabs"
            indicatorColor="secondary"
          >
            <Tab label="default-tab" sx={{ display: 'none' }} />
            {listCategories.map(({ category, endpoint, url }) => (
              <CustomTab
                label={category}
                key={category}
                onClick={() => {
                  handleClick({
                    urlParameter: category,
                    endpoint,
                    navigateUrl: url,
                  });
                }}
              />
            ))}
          </Tabs>
        </ThemeProvider>
      </Box>
    </Box>
  );
};
