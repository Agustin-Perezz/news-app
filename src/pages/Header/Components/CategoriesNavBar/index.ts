export * from './CategoriesNavBar';
export * from './customStyleTab';
export * from './listCategories';
