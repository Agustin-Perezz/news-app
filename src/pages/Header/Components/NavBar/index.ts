export * from './NavBar';
export * from './CustomStylesNavBar';
