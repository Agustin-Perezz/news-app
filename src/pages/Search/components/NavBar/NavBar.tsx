import { useContext } from 'react';
import { ImageListItem, Typography } from '@mui/material';
import { CustomStack } from '../../../../components/CustomStack';
import { SocialMedia } from '../../../../components/ui/SocialMedia';
import { SearchBar } from '../SearchBar';
import { NewsContext } from '../../../../context/NewsContext';

import news_logo from '../../../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  return (
    <CustomStack
      stackProps={{
        direction: 'row',
        justifyContent: 'space-evenly',
        alignItems: 'center',
        marginBottom: -1.5,
      }}
    >
      <ImageListItem sx={{ width: 190 }}>
        <img src={news_logo} alt="news-logo" />
      </ImageListItem>
      <SearchBar />
      <SocialMedia />
    </CustomStack>
  );
};
