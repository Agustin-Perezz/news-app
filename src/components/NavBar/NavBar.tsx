import { ImageListItem } from '@mui/material';
import useMediaQuery from '@mui/material/useMediaQuery';
import { SearchBar } from './SearchBar';
import { SocialMedia } from '../ui/SocialMedia';
import { CategoriesNavBar } from './CategoriesNavBar';
import { NavBarDrawer } from './Drawer';
import { CustomBoxContainer, CustomNavBarBox } from './CustomStyleNavBar';
import { CategoryProvider } from '../../context/CategoryContext';
import { useSearch } from '../../hooks';

import news_logo from '../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  const { handleSubmit } = useSearch();
  const matches = useMediaQuery('(max-width:1000px)');

  return (
    <CategoryProvider>
      <CustomBoxContainer>
        <CustomNavBarBox>
          {matches ? (
            <NavBarDrawer handleSubmit={handleSubmit} />
          ) : (
            <ImageListItem sx={{ maxWidth: '130px' }}>
              <img src={news_logo} alt="news-logo" />
            </ImageListItem>
          )}
          <SearchBar handleSubmit={handleSubmit} />
          {!matches && <SocialMedia />}
        </CustomNavBarBox>
        {!matches && <CategoriesNavBar handleClick={handleSubmit} />}
      </CustomBoxContainer>
    </CategoryProvider>
  );
};
