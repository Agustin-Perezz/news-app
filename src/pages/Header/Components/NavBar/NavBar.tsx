import Box from '@mui/material/Box';
import { ImageListItem } from '@mui/material';
import { SearchBar } from '../SearchBar';
import { CategoriesNavBar } from '../CategoriesNavBar';
import { CustomStack } from '../../../../components/CustomStack';
import { SocialMedia } from '../../../../components/ui/SocialMedia';
import { useSearch } from '../../hooks';

import news_logo from '../../../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  const { handleSubmit } = useSearch();
  return (
    <Box
      sx={{
        backgroundColor: '#292929',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: 'max-content',
        flexWrap: 'wrap',
        paddingTop: '20px',
      }}
    >
      <CustomStack
        stackProps={{
          direction: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          width: '1000px',
        }}
      >
        <ImageListItem sx={{ width: 150 }}>
          <img src={news_logo} alt="news-logo" />
        </ImageListItem>
        <SearchBar handleSubmit={handleSubmit} />
        <SocialMedia />
      </CustomStack>
      <CategoriesNavBar handleClick={handleSubmit} />
    </Box>
  );
};
