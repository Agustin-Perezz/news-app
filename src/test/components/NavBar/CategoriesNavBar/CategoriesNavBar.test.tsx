import { fireEvent, screen } from '@testing-library/react';
import { CategoriesNavBar, listCategories } from '../../../../components/NavBar/CategoriesNavBar';
import { renderWhitContext } from '../../../utils';

describe('Test in <CategoriesNavBar />', () => {
  const handleClick = jest.fn();
  function setup() {
    return renderWhitContext({
      cache: {},
      pathname: '/home',
      children: <CategoriesNavBar handleClick={handleClick} />,
    });
  }
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('should show all text categories', () => {
    setup();
    listCategories.shift();
    listCategories.forEach(({ category }) => {
      screen.getByText(category);
    });
  });
  it('should call handleClick when click any category', () => {
    setup();
    const parameters = {
      endpoint: '/top?api_token=undefined&categories=sports',
      navigateUrl: 'category/sports',
      urlParameter: 'sports',
    };

    const sportCategory = screen.getByRole('tab', { name: 'sports' });
    fireEvent.click(sportCategory);

    expect(handleClick).toHaveBeenCalled();
    expect(handleClick).toHaveBeenCalledWith({ ...parameters });
  });
});
