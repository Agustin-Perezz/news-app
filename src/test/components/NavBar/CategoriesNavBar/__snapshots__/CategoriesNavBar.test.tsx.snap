// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CategoriesNavBar /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiBox-root css-mhl23k"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="category-tabs"
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              default-tab
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              home
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              general
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              science
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              sports
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              business
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              health
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              entertainment
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              tech
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              politics
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              food
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1gnsuec-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              travel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-xyhd53-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
